### PR TITLE
[CVP-1467] - Step to apply ISV-supplied resources on test clusters

### PR DESCRIPTION
**Pull request by jsztuka**

### CODE REVIEW

**Comment by jsztuka** [comment]
Secrets were added to the both:
https://vault.ci.openshift.org/
and
https://selfservice.vault.ci.openshift.org/
with the same name kube-secrets.
Is there some place else that it needs to be added in order to be succesfully injected in CI?
@petr-muller

**Comment by petr-muller** [comment]
/retitle [CVP-1467] - Step to apply ISV-supplied resources on test clusters

**Comment by jsztuka** [comment]
For some reason curl cant reach pyxis domain:
`curl: (6) Could not resolve host: pyxis.engineering.redhat.com `

File: ci-operator/step-registry/optional-operators/cvp-common/apply-secrets/optional-operators-cvp-common-apply-secrets-commands.sh [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+REHEARSAL_PYXIS_URL="ospid-dc63077a-911b-4318-a3e3-4bea601d8e45"
+OO_PYXIS_URL="${OO_PYXIS_URL:-$REHEARSAL_PYXIS_URL}"
+
+GPG_KEY='/var/run/cvp-pyxis-gpg-secret/cvp-gpg.key' # Secret file which will be mounted by DPTP
+GPG_PASS='/var/run/cvp-pyxis-gpg-secret/cvp-gpg.pass' # Secret file which will be mounted by DPTP
+PKCS12_CERT='/var/run/cvp-pyxis-gpg-secret/cvp-dptp.cert' # Secret file which will be mounted by DPTP
+PKCS12_KEY='/var/run/cvp-pyxis-gpg-secret/cvp-dptp.key' # Secret file which will be mounted by DPTP
+
+ISV_PID=OO_PYXIS_URL # The ISV partner project ID
```
Comment (petr-muller) [inline ci-operator/step-registry/optional-operators/cvp-common/apply-secrets/optional-operators-cvp-common-apply-secrets-commands.sh:14]:
```suggestion
ISV_PID=$OO_PYXIS_URL # The ISV partner project ID
```

:trollface:

Comment (jsztuka) [reply ci-operator/step-registry/optional-operators/cvp-common/apply-secrets/optional-operators-cvp-common-apply-secrets-commands.sh]: Jesus christ...thanks!

File: ci-operator/step-registry/optional-operators/cvp-common/apply-secrets/optional-operators-cvp-common-apply-secrets-commands.sh [hunk 2]
```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+
+# Steps for extracting and applying the kube_secrets ISV parameter
+# Expects the standard Prow environment variables to be set
+
+REHEARSAL_INSTALL_NAMESPACE="!create"
+# Example of valid pyxis URL: https://pyxis.api.redhat.com/v1/projects/certification/pid/ospid-55159875-7eb1-41b0-bd91-0638b73536f1
```
Comment (dirgim) [inline ci-operator/step-registry/optional-operators/cvp-common/apply-secrets/optional-operators-cvp-common-apply-secrets-commands.sh:7]: nit: remove the example to the partner certification URL

File: ci-operator/step-registry/optional-operators/cvp-common/apply-secrets/optional-operators-cvp-common-apply-secrets-commands.sh [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+apiVersion: v1
+kind: Namespace
+metadata:
+  $NS_NAMESTANZA
+EOF
+    )
+fi
+
+# Creating file that contains namespace name
+echo "$OO_INSTALL_NAMESPACE" > "${SHARED_DIR}"/cvp-common-apply-secrets-namespace.txt
```
Comment (dirgim) [inline ci-operator/step-registry/optional-operators/cvp-common/apply-secrets/optional-operators-cvp-common-apply-secrets-commands.sh:44]: suggestion: make the filename more generic, something like `operator-install-namespace.txt` so it makes sense if we/others create the namespace in different steps.

File: ci-operator/step-registry/optional-operators/cvp-common/apply-secrets/optional-operators-cvp-common-apply-secrets-ref.yaml [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    requests:
+      cpu: 300m
+      memory: 300Mi
+  credentials:
+    - namespace: test-credentials
+      name: cvp-pyxis-gpg-secret
+      mount_path: /var/run/cvp-pyxis-gpg-secret
+  documentation: |-
+    Collect the project information from the supplied Pyxis project URL,
+    decrypt the encrypted kube_objects using stored CVP's GPG key and
```
Comment (dirgim) [inline ci-operator/step-registry/optional-operators/cvp-common/apply-secrets/optional-operators-cvp-common-apply-secrets-ref.yaml:18]: suggestion: 
```suggestion
    decrypt the encrypted kube_objects and
```

File: ci-operator/step-registry/optional-operators/subscribe/optional-operators-subscribe-commands.sh [hunk 5]
```diff
@@ -50,7 +50,9 @@ echo "OO_CHANNEL:           $OO_CHANNEL"
 echo "OO_INSTALL_NAMESPACE: $OO_INSTALL_NAMESPACE"
 echo "OO_TARGET_NAMESPACES: $OO_TARGET_NAMESPACES"
 
-if [[ "$OO_INSTALL_NAMESPACE" == "!create" ]]; then
+if [[ -f "${SHARED_DIR}/cvp-common-apply-secrets-namespace.txt" ]]; then
+    OO_INSTALL_NAMESPACE=$(cat $SHARED_DIR/cvp-common-apply-secrets-namespace.txt)
```
Comment (dirgim) [inline ci-operator/step-registry/optional-operators/subscribe/optional-operators-subscribe-commands.sh:54]: same suggestion as above: make the filename more generic, something like operator-install-namespace.txt so it makes sense if we/others create the namespace in different steps.

File: ci-operator/step-registry/optional-operators/cvp-common/apply-secrets/optional-operators-cvp-common-apply-secrets-commands.sh (hunk 2 above)
Comment (jsztuka) [reply ci-operator/step-registry/optional-operators/cvp-common/apply-secrets/optional-operators-cvp-common-apply-secrets-commands.sh]: Removed.

File: ci-operator/step-registry/optional-operators/cvp-common/apply-secrets/optional-operators-cvp-common-apply-secrets-commands.sh (hunk 3 above)
Comment (jsztuka) [reply ci-operator/step-registry/optional-operators/cvp-common/apply-secrets/optional-operators-cvp-common-apply-secrets-commands.sh]: Thanks for suggestion, I tend to assign too specific file names.

File: ci-operator/step-registry/optional-operators/cvp-common/apply-secrets/optional-operators-cvp-common-apply-secrets-ref.yaml (hunk 4 above)
Comment (jsztuka) [reply ci-operator/step-registry/optional-operators/cvp-common/apply-secrets/optional-operators-cvp-common-apply-secrets-ref.yaml]: Removed.

File: ci-operator/step-registry/optional-operators/subscribe/optional-operators-subscribe-commands.sh (hunk 5 above)
Comment (jsztuka) [reply ci-operator/step-registry/optional-operators/subscribe/optional-operators-subscribe-commands.sh]: Modified.

**Pull request merged**